### ISSUE TITLE
Unsupported modify charset from latin1 to utf8mb4

### ISSUE THREAD

**Issue by bestwoody** [pingcap/tidb#34008]
## Bug Report

<p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This can be easily simulated<br>MySQL<span> </span><span class="error">[test]</span>&gt; create table t2(a varchar(20) )CHARSET=latin1;<br>MySQL<span> </span><span class="error">[test]</span>&gt; insert into t2 values('&lt;some value&gt;');<br>MySQL<span> </span><span class="error">[test]</span>&gt; alter table t2 DEFAULT CHARACTER SET utf8mb4;<br>ERROR 8200 (HY000): Unsupported modify charset from latin1 to utf8mb4</p><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">If we create another table with charset=utf8mb4 and insert from t2. There is no error<br>MySQL<span> </span><span class="error">[test]</span>&gt; create table t3 (a varchar(20)) charset=utf8mb4;<br>MySQL<span> </span><span class="error">[test]</span>&gt; insert into t3 select * from t2;<br>Query OK, 1 row affected (0.01 sec)<br>MySQL<span> </span><span class="error">[test]</span>&gt; select * from t3;<br><ins>------</ins></p><div class="table-wrap" style="margin: 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

a
--

</div><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><ins>------</ins><br>It is not clear if the above insert would convert latin char to utf8 correctly.</p><p style="margin: 10px 0px 0px; padding: 0px; color: rgb(23, 43, 77); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, Oxygen, Ubuntu, &quot;Fira Sans&quot;, &quot;Droid Sans&quot;, &quot;Helvetica Neue&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Customer is aware of the latin support issue as documented and in<span> </span><a href="https://github.com/pingcap/tidb/issues/18955" class="external-link" target="_blank" rel="nofollow noopener" style="color: rgb(0, 82, 204); text-decoration: var(--aui-link-decoration); cursor: pointer;">https://github.com/pingcap/tidb/issues/18955</a></p>

### 1. Minimal reproduce step (Required)
MySQL [test]> create table t2(a varchar(20) )CHARSET=latin1;
MySQL [test]> insert into t2 values('<some value>');
MySQL [test]> alter table t2 DEFAULT CHARACTER SET utf8mb4;

### 2. What did you expect to see? (Required)
Query OK
### 3. What did you see instead (Required)
ERROR 8200 (HY000): Unsupported modify charset from latin1 to utf8mb4
### 4. What is your TiDB version? (Required)
v5.2.2

**Comment by wjhuang2016** [comment]
We don't have a plan to support it in a short time.

**Comment by bestwoody** [comment]
> We don't have a plan to support it in a short time.

this affects POC user.

**Comment by xiongjiwei** [comment]
```
MySQL [test]> create table t3 (a varchar(20)) charset=utf8mb4;
MySQL [test]> insert into t3 select * from t2;
```

this shall work

**Comment by Defined2014** [comment]
I don't think we have plans to support it in the near future, so closing this issue.
